File: Backend/Ready2RoadBackend/src/main/java/Ready2road/Persistenza/Model/Wallet.java
package Ready2road.Persistenza.Model;

import com.google.gson.JsonObject;

import java.math.BigDecimal;

public class Wallet {
    private Long id;
    private BigDecimal saldo;
    private BigDecimal puntiAcquisto;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public BigDecimal getSaldo() {
        return saldo;
    }

    public BigDecimal getPunti(){
        return puntiAcquisto;
    }

    public void setSaldo(BigDecimal saldo) {
        this.saldo = saldo;
    }

    public void setPuntiAcquisto(BigDecimal punti){
        this.puntiAcquisto = punti;
    }

    public JsonObject toJson(){
        JsonObject jsonObject=new JsonObject();
        jsonObject.addProperty("id",this.getId());
        jsonObject.addProperty("saldo",this.getSaldo());
        jsonObject.addProperty("puntiAcquisto", this.puntiAcquisto);
        return jsonObject;
    }
}

File: ready2road/src/app/model/Wallet.ts
export class  Wallet{
    protected id:number;
    protected saldo:number;
    constructor(json:any){
        this.id=json.id;
        this.saldo=json.saldo;
    }
    getId= (): number =>{
        return this.id;
    }
    getSaldo= (): number =>{
        return this.saldo;
    }
    setId= (id:number): void =>{
        this.id=id;
    }
    setSaldo= (saldo:number): void =>{
        this.saldo=saldo;
    }
}
